### ISSUE TITLE
need AWS_CROMWELL_OUTPUTS_GZ

### ISSUE THREAD

**Issue by notestaff** [broadinstitute/cromwell#4216]
We have a WDL task where the container override is over the 8K limit due to large value of AWS_CROMWELL_OUTPUTS.  Can you add the same _GZ mechanism as for AWS_CROMWELL_INPUTS?  Thanks!

**Comment by notestaff** [comment]
Maybe, better to upload the values of env vars to the s3 bucket, and just have one env var pointing at it?  Then length of container overrides won't be a problem.
@danbills @elerrch

**Comment by elerch** [comment]
Yes, that's a valid alternative approach @notestaff. The tradeoff here is that by using the environment variables, no additional permissions are required for the process running Cromwell. By using an S3 bucket, we gain the ability of unlimited output size, but also increase the amount of configuration and permissions required to successfully run Cromwell. I think a good enhancement would be to add the ability to pass these things through S3 and let the user decide which mechanism and tradeoffs they would like.

This said, removing the need for AWS_CROMWELL_OUTPUTS_GZ doesn't actually solve the 8k limit issue. In the testing for haplotype caller, some of the commands **themselves** are larger than 8k, even without considering environment variables. So, regardless of the environment variable compression, some solution is still needed for command text compression. I plan to file a couple PRs on the ECS Agent repo to get the conversation started with ECS and AWS Batch service teams on the limit and the mechanism being used to pass data between the services.

**Comment by notestaff** [comment]
If the command is put into an env var, can the command run in the container be just $MYVAR  ?

The process running Cromwell already needs read-only access to the bucket, to get the result code.   And the config is already done by CloudFormation or script.  I’d prefer one-time config to having to rewrite workflows.

**Comment by notestaff** [comment]
p.s. @elerch as an interim fix, it would help us a lot if you could implement AWS_CROMWELL_OUTPUTS_GZ .   I tried adding it (and rebuilding the custom AMI) but am finding that without a good understanding of the infrastructure it may take a while to debug.

**Comment by notestaff** [comment]
Just to check, this is not yet addressed in the recently release Release 36?

**Comment by ruchim** [comment]
Hello @notestaff 

This sounds like a good feature request, but not something that has been addressed yet and it might be a few months before we get to it. For now, is it possible that the output files can be bundled into a zipped file and passed onto a downstream task possibly? As an attempt to consolidate the outputs.

Thanks!

**Comment by notestaff** [comment]
@ruchim Thanks for the update.  I can find a workaround, e.g. reducing the number of exposed outputs; it's just that this will affect others using the same workflow on e.g. DNAnexus .  If I try to make a PR myself, based on what's done for AWS_CROMWELL_INPUTS_GZ, do you think you'd be able to look at it?

**Comment by ruchim** [comment]
@notestaff Absolutely!
